.github/workflows/regression-tests.yml

```diff
@@ -135,6 +135,8 @@ jobs:
 
     steps:
     - uses: actions/checkout@v3
+      with:
+        fetch-depth: 2
```

Comment:
Can you please add a comment as to why this is necessary? We had fetch-depth in there previously for setuptools_scm, and I removed it because we don't use setuptools_scm any more. Now we need it because of codecov, which isn't obvious by looking at that line.